rust/transaction_parsing/src/tests.rs

```diff
@@ -282,85 +696,475 @@ fn parse_transaction_1() {
+                        period: "64".to_string(),
+                    },
+                },
+            },
+            TransactionCard {
+                index: 8,
+                indent: 0,
+                card: Card::NonceCard {
+                    f: "89".to_string(),
+                },
+            },
+            TransactionCard {
+                index: 9,
+                indent: 0,
+                card: Card::TipCard {
+                    f: MSCCurrency {
+                        amount: "300.000000000000".to_string(),
+                        units: "WND".to_string(),
+                    },
+                },
+            },
+            TransactionCard {
+                index: 10,
+                indent: 0,
+                card: Card::NameVersionCard {
+                    f: MSCNameVersion {
+                        name: "westend".to_string(),
+                        version: "9010".to_string(),
+                    },
+                },
+            },
+            TransactionCard {
+                index: 11,
+                indent: 0,
+                card: Card::TxSpecCard { f: "5".to_string() },
+            },
+            TransactionCard {
+                index: 12,
+                indent: 0,
+                card: Card::BlockHashCard {
+                    f: "a8dfb73a4b44e6bf84affe258954c12db1fe8e8cf00b965df2af2f49c1ec11cd"
+                        .to_string(),
+                },
+            },
+        ]),
+        ..Default::default()
+    };
+
+    let author_info_known = Address {
+        base58: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY".to_string(),
+        identicon: alice_sr_alice().to_vec(),
+        seed_name: "Alice".to_string(),
+        path: "//Alice".to_string(),
+        has_pwd: false,
+        multiselect: None,
+    };
+    let network_info_known = westend_spec();
     let output = produce_output(line, dbname);
-    if let Action::Sign {
+    if let TransactionAction::Sign {
         content,
         checksum: _,
         has_pwd,
         author_info,
         network_info,
     } = output
     {
-        let author_info_cut = author_info.replace(&alice_sr_alice(), r#"<alice_sr25519_//Alice>"#);
-        let content_cut = content.replace(&bob(), r#"<bob>"#);
-        assert!(content_cut == content_known, "Received: \n{}", content);
-        assert!(
-            author_info_cut == author_info_known,
-            "Received: \n{}",
-            author_info
-        );
-        assert!(
-            network_info == network_info_known,
-            "Received: \n{}",
-            network_info
-        );
+        assert_eq!(content, content_known);
+        assert_eq!(author_info, author_info_known);
         assert!(!has_pwd, "Expected no password");
+        assert_eq!(network_info, network_info_known);
     } else {
         panic!("Wrong action {:?}", output)
     }
     fs::remove_dir_all(dbname).unwrap();
 }
 
+/* TODO: does this actually test anything meaningful?
```

Comment:
Not anymore, I suppose. This was mostly to test for punctuation errors in json. Off it goes?
We still need some test to see that all cards are displayed as expected on screen though.